app/main.py

```diff
@@ -1,3 +1,17 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable, Any
+import functools
+
+
+def cache(func: Callable) -> Any:
+    memory = {}
+
+    @functools.wraps(func)
+    def inner(*args) -> Any:
```

Comment:
Add here `**kwargs` and pass them to the `func` as well. Their processing is not needed for this task, but presence would be preferable.